### ISSUE TITLE
[Feature] row and col option as function

### ISSUE THREAD

**Issue by ksk0** [akinsho/toggleterm.nvim#343]
I'm using floating terminal window, with dynamically defined size accounting for cmdline height with config:

```
local config = {
  float_opts = {
    width = function()
      return math.floor(vim.o.columns * 0.9)
    end,

    height = function()
      return math.floor((vim.o.lines - vim.o.cmdheight) * 0.9)
    end,
  }
}

require("toggleterm").setup{config}
```
While this, defines window of appropriate size, it's placement is in the center of screen according to formula:
```
row = math.ceil(vim.o.columns - width) / 2 - 1 
```
resulting with  command line area overlapping the floating window. Correct row value should be:
```
row = math.floor(0.05 * (vim.o.lines - vim.o.cmdheight))
```
modifying config as:
```
local config = {
  float_opts = {
    width = function()
      return math.floor(vim.o.columns * 0.9)
    end,

    height = function()
      return math.floor((vim.o.lines - vim.o.cmdheight) * 0.9)
    end,

    row = function()
      return math.floor(0.05 * (vim.o.lines - vim.o.cmdheight))
    end,
  }
}
```
does not yield  result since `toggleterm/ui.lua` does not accept `row` and `col` as functions (error is raised). I also tried with metatable:
```
setmetatable(config.float_opts, {
  __index = function(_, key)
    if key == "row" then
      return math.floor(0.05 * (vim.o.lines -  vim.o.cmdheight))
    end
  end
})
```
but that also does not yield a result, since in function `M.set()` of `toggleterm/config.lua`  during "deep_extend" metatable is lost:
```
config = vim.tbl_deep_extend("force", config, user_conf)
```
Would it be possible to modify the code in `toggleterm/ui.lua` from:
```
row = (opts.row or math.ceil(vim.o.lines - height) / 2) - 1,
```
to:
```
row = M._resolve_size(opts.row, term) or ((math.ceil(vim.o.lines - height) / 2) - 1)
```
which would result in ability do define `row` and `col` as functions

**Comment by akinsho** [comment]
@ksk0 I'd take a PR to add this functionality, but please ensure before raising it that `_resolve_size` is a general enough solution to use 🙏🏿

**Comment by akinsho** [comment]
@ksk0 I think something like
```lua
row = vim.F.if_nil(M.resolve_size(opts.row, term), default_value)
```
would be better

**Comment by sunn4room** [comment]
It's confuse to define a function for each item of 'float_opts', I think it would be useful to add an option 'float_opts_fn', If 'float_opts_fn' is a function, use this function, otherwise use 'float_opts'.

**Comment by akinsho** [comment]
@sunn4room this is not necessary since a user can provide a value *or a function* you don't have to use a function if you don't want to

**Comment by akinsho** [comment]
fixed by #347
